src/later.cpp

```diff
@@ -46,7 +46,12 @@ int sys_nframe() {
 bool at_top_level() {
   if (exec_callbacks_reentrancy_count != 0)
     return false;
-  return sys_nframe() == 0;
+
+  int nframe = sys_nframe();
+  if (nframe == -1) {
+    throw Rcpp::exception("Error occurred while calling sys.nframe()");
```

Comment:
Could also make a case for returning false here. Though I guess it's a good idea at least for now, so we can see if it ever happens.